mlflow/pyfunc/__init__.py

```diff
@@ -357,6 +372,8 @@ def add_to_model(
             params[ENV][EnvType.CONDA] = conda_env
         if python_env:
             params[ENV][EnvType.VIRTUALENV] = python_env
+    if model_config:
+        params[MODEL_CONFIG] = model_config
     return model.add_flavor(FLAVOR_NAME, **params)
```

Comment:
Instead of saving load_config in params, could we add an attribute for Model called `load_config` directly, and save load_config when saving/logging the model? e.g. https://docs.google.com/document/d/11sRxfUIXsdsQkE_7bZTviK0fFyhP9PNIr2dqrIT05ew/edit#heading=h.ggqwwxf0svm

Reply:
As we discussed yesterdat, the proposal considers model configuration for the `pyfunc` flavor. Adding it to the root of model will imply this is something available to all the flavors, which is not the case because it will force all the flavors to follow the same pattern, which is an unnecessary enforcement. Does it make sense?

Reply:
We should only add `load_config` to flavors we would like to support. For example, in sklearn flavor we won't add `load_config` in its save_model or log_model function, then the attribute will just be None for those models.

Reply:
+1 to @serena-ruan 's suggestion. I don't think that it'll be an all or nothing sort of deal. There are flavors that don't have the concept of providing these on-load parameters, which is fine. Provided that the logic is applied such that flavors that are not supportive of any overrides to loading configuration aren't required to have the user define anything (and not to expose an arg on load for this behavior that would effectively be a no-op), it will help to enable more use cases that are not strictly related to model serving / using a pyfunc representation of the model. Being able to override for static batch processing is likely to be a worthwhile feature.

Reply:
@santiagxf I think you could take another look at my design doc, and we could discuss on the design doc first if there're anything you would like to change :D Then we can implement according to the design doc.

Reply:
@serena-ruan, let me suggest doing it the other way around. This PR follows the original doc that we circulated for the feature so it doesn't make sense to have a discussion over a document that describes different approaches. What do you think about that?

Reply:
The initial doc was for params and load_configs that are mixed together. That doc I wrote is for load_config specifically, now that we already have params for inference, we should follow the new doc ( and we already had internal discussion on the design doc).

Reply:
@serena-ruan I have some concerns about the new doc's use of `**kwargs`. We shouldn't use **kwargs for load-time config passed to `mlflow.pyfunc.load_pyfunc()` because it's very likely to lead to future compatibilities. E.g. say we want to introduce a new argument to mlflow.pyfunc.load_pyfunc() in the future called `environment` that lets users load the model in a virtualenv / conda env; we wouldn't be able to safely add that arg because someone might have a model config called `environment`. **kwargs are not future-proof; I prefer @santiagxf 's implementation here.

Regarding adding `model_config` to the pyfunc flavor or to the top-level model config, I'm again aligned with @santiagxf: `params[MODEL_CONFIG]` makes sense. If we put the model config at the top level, it means that the flavor-specific `load_model()` implementations would all have to support `model_config`. This would create confusion with existing APIs like `mlflow.pytorch.load_model()` which accept arguments like `device`: should users pass `device` or `model_config`? IMO, flavor-specific `load_model()` APIs should take **kwargs or enumerate the framework-specific load-time args that they support, and pyfunc should use a more generic `model_config` dict, which is exactly what @santiagxf's PR does.

Can we move forward with the current PR as-is?